extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -128,6 +129,16 @@ function SidebarWithRef(
       return;
     }
 
+    const document = sidebarElement.ownerDocument;
+    if (side != Side.LEFT && side != Side.RIGHT) {
+      sideRef.current = isRTL(document) ? Side.RIGHT : Side.LEFT;
+    } else {
+      sideRef.current = side;
+    }
```

Comment:
```suggestion
    sideRef.current = isEnumValue(Side, side) && side || isRTL(document) ? Side.RIGHT : Side.LEFT;
```

or 
```suggestion
    if (!isEnumValue(Side, side)) {
        // throw ?
    }
    sideRef.current = side ?? isRTL(document) ? Side.RIGHT : Side.LEFT;
```

I only suggest the ladder because typos may fail silently, i.e. `side="rihgt"` for ltr documents. But maybe that is reasonable. 🤷‍♀️  In AMP document validations we would normally be able to assert that the only values for the attrs could be `left|right`.

Reply:
Thanks, I have added in `isEnumValue`.  I think we can fail silently for something like `side='rihgt'`.  This case will just default us to either `right` or `left` based on whether or not the document is `RTL` which I think is reasonable.

Reply:
I think the decision we worked out is that Preact components do not explicitly fail on illegal inputs. But if they happen to fail - it's ok. In other words: we don't pay much attention to the input in Preact: just document the values in `type.js` files and proceed assuming the input is correct.